### ISSUE TITLE
`mjml-cli`: track custom components as dependencies when using `watch`

### ISSUE THREAD

**Issue by IanEdington** [mjmlio/mjml#2025]
**Is your feature request related to a problem? Please describe.**
Currently the `watch` flag in mjml's cli builds a dependency tree of mjml partials. However, custom components are not tracked. It would be valuable for components to trigger a rebuild.

**Describe the solution you'd like**
Track components as a file dependencies. Reload component and rebuild on change.

**Describe alternatives you've considered**
The alternative is either a slow gulp process that doesn't consider dependencies, or manually restarting the watch when components are changed.

**Additional context**
Since it doesn't seem like this has been brought up in a previous issue, I mainly want to record it for the future. It isn't currently a bottleneck, however, any suggestion on how to get this working would be appreciated. 🙏

**Comment by iRyusa** [comment]
I don't know if we want to do such thing as it a bit tricky to hot reload stuff like component

**Comment by IanEdington** [comment]
I can see that reasoning. I haven't dug too deeply into the internals of how the cli is doing the watch operation. I imagined running the mjml renderer as a subprocess and reloading the subprocess when a component changes.

I mostly put in this issue to track it. This isn't a large blocker for me at the moment since I've set up two processes in gulp, one for template development and the other for component development. It seems to be working well.
